lib/ui/painting/fragment_shader.cc

```diff
@@ -48,4 +49,78 @@ FragmentShader::FragmentShader(
 
 FragmentShader::~FragmentShader() = default;
 
+IMPLEMENT_WRAPPERTYPEINFO(ui, ReusableFragmentShader);
+
+ReusableFragmentShader::ReusableFragmentShader(
+    fml::RefPtr<FragmentProgram> program,
+    uint64_t float_count,
+    uint64_t sampler_count)
+    : program_(program),
+      uniform_data_(SkData::MakeUninitialized(
+          (float_count + 2 * sampler_count) * sizeof(float))),
+      samplers_(sampler_count),
+      float_count_(float_count) {}
+
+Dart_Handle ReusableFragmentShader::Create(Dart_Handle wrapper,
+                                           Dart_Handle program,
+                                           Dart_Handle float_count_handle,
+                                           Dart_Handle sampler_count_handle) {
+  auto* fragment_program =
+      tonic::DartConverter<FragmentProgram*>::FromDart(program);
+  uint64_t float_count =
+      tonic::DartConverter<uint64_t>::FromDart(float_count_handle);
+  uint64_t sampler_count =
+      tonic::DartConverter<uint64_t>::FromDart(sampler_count_handle);
```

Comment:
nit: if you type these in the declaration tonic will just do this for you.

Reply:
That's how I had it at first, but the integer parameters were getting the value of the `Dart_Handle` rather than the value wrapped by the `Dart_Handle`. I'm going to leave these as is, and try to make a repro of the issue I was seeing.